### PR TITLE
feat(model): flux2

### PR DESCRIPTION
**Pull request by linnanwang**
# FLUX.2-dev End-to-End Finetuning & Inference PR Summary

## What This PR Does

This PR adds end-to-end support for **FLUX.2-dev**, a text-to-image diffusion model that uses:

- A **4D positional ID scheme**
- **Mistral3-based text embeddings**
- Flow-matching finetuning
- Full-finetune and LoRA training workflows
- Image generation/inference support

In short, it makes FLUX.2-dev usable inside the diffusion training and inference stack: preprocess data, finetune the model, and generate images from the trained checkpoint.

---

### 1. `Flux2Adapter`

**File:** `nemo_automodel/components/flow_matching/adapters/flux2.py`

Adds a new `ModelAdapter` subclass for FLUX.2-dev.

#### Responsibilities

- Accepts patchified and BatchNorm-normalized latents with shape:

### CODE REVIEW

File: nemo_automodel/components/flow_matching/adapters/flux2.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+from typing import Any, Dict
+
+import torch
+import torch.nn as nn
+
+from .base import FlowMatchingContext, ModelAdapter
+
+
+class Flux2Adapter(ModelAdapter):
```
[the remark this answers is not in the document]
Comment (linnanwang) [reply nemo_automodel/components/flow_matching/adapters/flux2.py:34]: Added in commit `0a862a37`: `tests/unit_tests/flow_matching/test_flux2_adapter.py` — 36 tests covering pack/unpack roundtrip + contiguity, 4D `img_ids`/`txt_ids` shape/dtype/value correctness, `prepare_inputs` output keys, CFG dropout, timestep normalization, and `forward` call kwargs.

File: tools/diffusion/processors/flux2.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        return "black-forest-labs/FLUX.2-dev"
+
+    def load_models(self, model_name: str, device: str) -> Dict[str, Any]:
+        """
+        Load FLUX.2 models from Flux2Pipeline.
+
+        Args:
+            model_name: HuggingFace model path (e.g., 'black-forest-labs/FLUX.2-dev')
+            device: Device to load models on
+
```
[the remark this answers is not in the document]
Comment (linnanwang) [reply tools/diffusion/processors/flux2.py:66]: Added in commit `0a862a37`: `tests/unit_tests/diffusion_processors/test_flux2_processor.py` — 22 tests covering `model_type`/`default_model_name` properties, `encode_image` BN normalization + dtype + batch-dim squeeze, `encode_text` Mistral3 args + no-CLIP-key assertions, `verify_latent` shape/NaN/Inf checks (no VAE decode needed), `get_cache_data` structure, and `ProcessorRegistry` lookup.

**Review by akoumpa — APPROVED** [review]
